docs/adv/troubleshooting/errors.md

```diff
@@ -0,0 +1,328 @@
+
+### How do I get my ENR if I want to generate it again?
+
+`cd` to the directory where your private keys are located (ex: `cd /path/to/charon/enr/private/key`)
+
+Run `docker run --rm -v "$(pwd):/opt/charon" obolnetwork/charon:v1.2.0 enr`. This prints the ENR on your screen.
+
+Please note that this ENR is not the same as the one generated when you created it for the first time. This is because the process of generating ENRs includes the current timestamp.
+
+
+### What do I do if lose my `charon-enr-private-key`?
+
+For now, ENR rotation/replacement is not supported, it will be supported in a future release. Therefore, its advised to always keep a backup of your `charon-enr-private-key ` in a secure location (ex: cloud storage, USB Flash drive, etc.).
+
+### I can`t find the keys anywhere
+The `charon-enr-private-key` is generated inside a hidden folder `.charon`. To view it, run `ls -al` in your terminal. You can then copy the key to your `~/Downloads` folder for easy access by running `cp .charon/charon-enr-private-key ~/Downloads`. This step may be a bit different for Windows.
+Else, if you are on macOS, press `Cmd + Shift + .` to view the `.charon` folder in the Finder application.
+
+
+## Lighthouse
+
+
+### Lighthouse says "downloading historical blocks"
+
+This means that Lighthouse is still syncing which will throw a lot of errors down the line. Wait for the sync before moving further.
+
+
+### Lighthouse gives the error `failed to request attester duties`
+
+This indicates there is something wrong with your Lighthouse beacon node. This might be because the request buffer is full as your node is never starting consensus since it never gets the duties.
+
+
+### Lighthouse gives the error `not enough time for a discovery seach`
+
+This could be linked to a internet connection being too slow or relying on a slow third-party service such as Infura.
+
+
+## Beacon Node
+
+### `Error communicating with Beacon Node API` & `Error while connecting to beacon node event stream`
+
+This is likely due to Lighthouse not done syncing, wait and try again once synced. Can also be linked to Teku keystore issue.
+
+### Clock sync issues
+Either your clock server time is off, or you are talking to a remote beacon client that is super slow (this is why we advise against using services like Infura).
+
+
+### My beacon node API is flaky with lots of errors and timeouts
+
+A good quality beacon node API is critical to validator performance. It is always advised to run your own beacon node to ensure low latencies to boost validator performance.
+Using 3rd party services like Infura`s beacon node API has significant disadvantages since the quality is often low. Requests often return 500s or timeout (Charon times out after 2s). This results in lots of warnings and errors and failed duties. Running a local beacon node is always preferred.
+We are not yet considering increasing the 2s timeout since that can have knock-on effects.
```

Comment:
This is customisable now.

```suggestion
Using 3rd party services like Infura's beacon node API has significant disadvantages since the quality is often low. Requests often return 500s or timeout. This results in lots of warnings and errors and failed duties. Running a local beacon node is always preferred.
```